### PR TITLE
MacOS code signing support

### PR DESCRIPTION
**Pull request by johnalotoski**
This PR adds MacOS code signing support.

## Todos

- [ ] Todo

## Screenshots

- Screenshot

---

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/messages/GGKFXSKC6)
- [ ] Test

---

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

File: installers/common/MacInstaller.hs [hunk 1]
```diff
@@ -96,6 +114,39 @@ main opts@Options{oBackend, oCluster, oBuildJob, oOutputDir, oTestInstaller, oSi
...
+eval "$SIGN_CMD \"$ABS_PATH/Contents/Frameworks/Electron Framework.framework/Versions/Current/Resources/crashpad_handler\" $LOG"
+eval "$SIGN_CMD \"$ABS_PATH/Contents/Frameworks/Electron Framework.framework/Versions/Current/Libraries/libnode.dylib\" $LOG"
+eval "$SIGN_CMD \"$ABS_PATH/Contents/Frameworks/Electron Framework.framework/Versions/Current/Libraries/libffmpeg.dylib\" $LOG"
+
+# Sign the whole component deeply
+eval "$SIGN_CMD \"$ABS_PATH\" $LOG"
+eval "$VERIFY_CMD \"$ABS_PATH\" $LOG"
+eval "$VERIFY_CMD --display -r- \"$ABS_PATH\""
+set +x|]
+
```
Comment (cleverca22) [inline installers/common/MacInstaller.hs]: maybe this should be a shell script that nix just adds to `$PATH` for us?

File: installers/common/MacInstaller.hs [hunk 2]
```diff
@@ -260,15 +311,40 @@ writeLauncherFile dir DarwinConfig{dcDataDir,dcAppName} = do
       , "\"$(dirname \"$0\")/cardano-launcher\""
       ]
 
+writeCodesignScriptFile :: String -> IO FilePath
+writeCodesignScriptFile codesignScriptContents = do
+  writeTextFile codesignScript $ T.pack codesignScriptContents
+  run "chmod" [ "+x", codesignScriptRef ]
```
Comment (cleverca22) [inline installers/common/MacInstaller.hs]: https://hackage.haskell.org/package/turtle/docs/Turtle-Prelude.html#v:chmod

File: installers/common/MacInstaller.hs (hunk 1 above)
Comment (johnalotoski) [reply installers/common/MacInstaller.hs]: I'm ok leaving here from now and moving it out if we add any additional complexity in the future.

**Comment by nikolaglumac** [comment]
@alan-mcnicholas we depend on your approval here since you are the only QA team member that has a mac 🙏 
cc @cleverca22 @johnalotoski

**Comment by nikolaglumac** [comment]
Closing this PR unmerged as this fix has been already merged into `release/2.3.1-ITN1` branch which is to be merged into `v2-integration` today: https://github.com/input-output-hk/daedalus/pull/1890

**Pull request closed without merging**